src/BTree.h

```diff
@@ -1144,13 +1153,19 @@ class btree {
     using operation_hints = btree_operation_hints<1>;
 
 private:
-#if defined(IS_PARALLEL) && !defined(HAS_TSX)
+#ifdef IS_PARALLEL
+    // the total number of elements in this tree
+    volatile size_type numElements;
+
     // a pointer to the root node of this tree
     node* volatile root;
 
     // a lock to synchronize update operations on the root pointer
     lock_type root_lock;
 #else
+    // the total number of elements in this tree
+    size_type numElements;
```

Comment:
You've probably thought of this, but just to check - do we really check the size often enough for the overhead of counting to be worthwhile?